chanbackup/backup.go

```diff
@@ -53,11 +56,44 @@ func assembleChanBackup(addrSource AddressSource,
 	return &single, nil
 }
 
+func buildCloseTx(targetChan *channeldb.OpenChannel, signer input.Signer) ([]byte, error) {
+	chanMachine, err := lnwallet.NewLightningChannel(signer, targetChan, nil)
+	if err != nil {
+		return nil, fmt.Errorf("lnwallet.NewLightningChannel failed: %w", err)
+	}
+	closeSummary, err := chanMachine.GetLocalForceCloseSummary()
```

Comment:
This will return a signed force close transaction. But I think for security reasons we should NOT have the signed TX in the SCB. Because that way it would be too easy for someone to decode the SCB, take the TX and publish it, breaching themselves.
I think we should only make it possible to extract and sign the TX using the appropriate `chantools` command, where we can show all the warnings and ask for confirmation several times...

Reply:
I see two possible approaches how to address the security concerns.

1. Store two separate fields in SCB: unsigned transaction and encrypted signature of it. In this case the transaction is unlikely to be extracted from SCB and published accidentally, because it has to be decrypted and reassembled.
2. Actually store in SCB all the inputs except the signer. To do that, the logic of signing should be factored out from the code of [LightningChannel.getSignedCommitTx](https://github.com/lightningnetwork/lnd/blob/v0.17.3-beta/lnwallet/channel.go#L6249-L6358) and missing inputs should be added to SCB. Fortunately, some of them are already in SCB or are easy to add. The most problematic input is [input.SignDescriptor](https://pkg.go.dev/github.com/lightningnetwork/lnd@v0.17.3-beta/input#SignDescriptor) which has interface field PrevOutputFetcher. Encode and decode code is needed for these types to store them in SCB. If new channel type appears, new variables will be needed in SCB to produce commit tx.

I'm inclining to the first approach, because it is sufficient to improve security of SCB with regard to accidental transaction publishing, but it is isolated in the code base and does not add maintenance burden for future channel types.

What do you think?

Reply:
To create a signed force close transaction, the following additional information is needed in SCB:

 * CommitTx (unsigned)
 * CommitSig
 * CommitHeight (for taproot channels only)

All other inputs seem to be already in SCB file.

I'll test how it works altogether in lnd & chantools.

Reply:
I implemented the changes for the second approach. It works! I produced valid transactions in testnet using `chantools scbforceclose`.